core/src/main/java/com/expedia/www/haystack/client/dispatchers/ChainedDispatcher.java

```diff
@@ -13,6 +13,10 @@
 public class ChainedDispatcher implements Dispatcher {
     private final List<Dispatcher> dispatchers;
 
+    public ChainedDispatcher(List<Dispatcher> dispatchers) {
+        this.dispatchers = Collections.unmodifiableList(dispatchers);
+    }
+
     public ChainedDispatcher(Dispatcher... dispatchers) {
```

Comment:
Shouldn't all constructors put their data in an unmodifiable list, the way that the new constructor does?

Reply:
`Array` vs. `Collection` semantics

Reply:
I will change the `Array` constructor to me an unmodifiable list.